### PR TITLE
1060 format work

### PR DESCRIPTION
**Pull request by geissonator**
These are the changes required to get phosphor-network working with the latest clang17 and latest yocto changes that we want to put into 1060-ghe.

### CODE REVIEW

File: src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp [hunk 1]
```diff
@@ -1003,7 +1003,7 @@ HypEthInterface::DHCPConf HypEthInterface::dhcpEnabled(DHCPConf value)
         bool v4Enabled = false;
         bool v6Enabled = false;
         bool slaacEnabled = false;
-        HypEthernetIntf::DHCPConf newValue;
+        HypEthernetIntf::DHCPConf newValue = HypEthernetIntf::DHCPConf::v4;
```
Comment (raviteja-b) [inline src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp:1006]: Need to test if this change has any impact on VMI network configuration

**Review by raviteja-b — APPROVED** [review]

File: src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp (hunk 1 above)
Comment (sunharis) [inline src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp:1006]: Was this needed for the format work?

Comment (raviteja-b) [reply src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp:1006]: seems its needed https://github.com/ibm-openbmc/phosphor-networkd/pull/116/commits/a3d268447e3dbfa9ffa4f64344843a17b9775dc5

Comment (sunharis) [reply src/ibm/hypervisor-network-mgr-src/hyp_ethernet_interface.cpp:1006]: Sure. Thanks Andrew. I think this should be OK. 
But we will keep an eye on the same and will need @prkatti1 to run the VMI network test regression bucket after this.

**Review by sunharis — APPROVED** [review]

**Pull request merged**